internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1848,7 +1848,7 @@ func resourceKubernetesClusterCreate(d *pluginsdk.ResourceData, meta interface{}
 	if maintenanceConfigRaw, ok := d.GetOk("maintenance_window_auto_upgrade"); ok {
 		client := meta.(*clients.Client).Containers.MaintenanceConfigurationsClient
 		parameters := maintenanceconfigurations.MaintenanceConfiguration{
-			Properties: expandKubernetesClusterMaintenanceConfiguration(maintenanceConfigRaw.([]interface{})),
+			Properties: expandKubernetesClusterMaintenanceConfiguration(maintenanceConfigRaw.([]interface{}), nil),
```

Comment:
Instead of passing in nil here could you please split the expand into two separate functions, one for create and one for update so it's clear at first glance that they are being handled differently?

Reply:
Thanks @stephybun ! I've updated this PR as suggested.